### ISSUE TITLE
Deploy PyDash.io on external server

### ISSUE THREAD

**Issue by TheVeggydude** [RUGSoftEng/2018-PyDash.io#160]
- [ ] Deploy the entire application to the server.
- [ ] Set up automated deployment (?)

**Comment by Qqwy** [comment]
We are still waiting for @mircealungu on this.

**Comment by InputUsername** [comment]
Some things to note when deploying:
- Not using the Flask built in WSGI server, it's easy but does not scale well;
- Using a web server such as Nginx (preferable) or Apache to serve static content like the React app, using Flask itself for that would incur extra overhead;
- http://flask.pocoo.org/docs/1.0/quickstart/#deploying-to-a-web-server contains useful info about deployment.

**Comment by Qqwy** [comment]
OK. pydash runs properly on pydash.io, including when the server is restarted
It would only be nice if someone set up proper github credentials for the pydash user
so we can properly pull and also push from there
(I've made a couple of changes to `build_and_run.sh` on the server to allow running in production mode)
To check how the server is doing, run `sudo service pydash status`
(starting/stopping is done with `sudo service pydash start` and `sudo service pydash stop`. It is auto-started at server startup)
The systemd config can be found in `/etc/systemd/system/pydash.service`. It might be improved somewhat (the PATH env is hard-coded in there because that was the only way I got it to find the proper executables)
The nginx config file is found in `/etc/nginx/sites-available/pydash.io.conf`, which straightforwardly proxies to localhost:5000 (and redirects http to https)
